activejob/lib/active_job/test_helper.rb

```diff
@@ -10,6 +10,7 @@ module TestHelper
 
     def before_setup # :nodoc:
       test_adapter = queue_adapter_for_test
+      self.class.to_s.gsub(/Test\z/, "").safe_constantize
```

Comment:
I'd write this as `self.class.name.chomp('Test').safe_constantize`.

What happens if users call `tests SomethingUnconventional`?
I found the original issue tough to grasp, but is this the best way we can find to solve the issue?